Exists filter throws NullPointerException on empty index

In 1.7.1 
Using Filtered query with Exists filter causes SearchParseException on an empty index.

Steps to reproduce:
````
# make sure there is no such index
curl -XDELETE localhost:9200/test_index

# create a new index
curl -XPUT localhost:9200/test_index

# issue a query with Exists Filter
curl -XGET localhost:9200/test_index/_search -d '
{
    "query": {
        "filtered": {
           "query": {
           "match_all": {}
           },
           "filter": {
               "exists": {
                  "field": "user"
               }
           }
        }
    }
}
'
````
Stack trace:
````
org.elasticsearch.search.SearchParseException: [test_index][3]: from[-1],size[-1]: Parse Failure [Failed to parse source [{
    "query": {
        "filtered": {
           "query": {
           "match_all": {}
           },
           "filter": {
               "exists": {
                  "field": "user"
               }
           }
        }
    }
}
]]
	at org.elasticsearch.search.SearchService.parseSource(SearchService.java:747)
	at org.elasticsearch.search.SearchService.createContext(SearchService.java:572)
	at org.elasticsearch.search.SearchService.createAndPutContext(SearchService.java:544)
	at org.elasticsearch.search.SearchService.executeQueryPhase(SearchService.java:306)
	at org.elasticsearch.search.action.SearchServiceTransportAction$5.call(SearchServiceTransportAction.java:231)
	at org.elasticsearch.search.action.SearchServiceTransportAction$5.call(SearchServiceTransportAction.java:228)
	at org.elasticsearch.search.action.SearchServiceTransportAction$23.run(SearchServiceTransportAction.java:559)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
	at java.lang.Thread.run(Thread.java:745)
Caused by: java.lang.NullPointerException
	at org.elasticsearch.index.query.ExistsFilterParser.newFilter(ExistsFilterParser.java:87)
	at org.elasticsearch.index.query.ExistsFilterParser.parse(ExistsFilterParser.java:82)
	at org.elasticsearch.index.query.QueryParseContext.executeFilterParser(QueryParseContext.java:371)
	at org.elasticsearch.index.query.QueryParseContext.parseInnerFilter(QueryParseContext.java:352)
	at org.elasticsearch.index.query.FilteredQueryParser.parse(FilteredQueryParser.java:74)
	at org.elasticsearch.index.query.QueryParseContext.parseInnerQuery(QueryParseContext.java:305)
	at org.elasticsearch.index.query.IndexQueryParserService.innerParse(IndexQueryParserService.java:382)
	at org.elasticsearch.index.query.IndexQueryParserService.parse(IndexQueryParserService.java:281)
	at org.elasticsearch.index.query.IndexQueryParserService.parse(IndexQueryParserService.java:276)
	at org.elasticsearch.search.query.QueryParseElement.parse(QueryParseElement.java:33)
	at org.elasticsearch.search.SearchService.parseSource(SearchService.java:731)
	... 9 more
````

--- a/src/main/java/org/elasticsearch/index/query/ExistsFilterParser.java
+++ b/src/main/java/org/elasticsearch/index/query/ExistsFilterParser.java
@@ -84,7 +84,7 @@ public class ExistsFilterParser implements FilterParser {
 
     public static Filter newFilter(QueryParseContext parseContext, String fieldPattern, String filterName) {
         final FieldMappers fieldNamesMappers = parseContext.mapperService().indexName(FieldNamesFieldMapper.NAME);
-        final FieldNamesFieldMapper fieldNamesMapper = (FieldNamesFieldMapper)fieldNamesMappers.mapper();
+        final FieldNamesFieldMapper fieldNamesMapper = fieldNamesMappers == null ? null : (FieldNamesFieldMapper)fieldNamesMappers.mapper();
 
         MapperService.SmartNameObjectMapper smartNameObjectMapper = parseContext.smartObjectMapper(fieldPattern);
         if (smartNameObjectMapper != null && smartNameObjectMapper.hasMapper()) {

--- a/src/main/java/org/elasticsearch/index/query/MissingFilterParser.java
+++ b/src/main/java/org/elasticsearch/index/query/MissingFilterParser.java
@@ -98,7 +98,7 @@ public class MissingFilterParser implements FilterParser {
         }
 
         final FieldMappers fieldNamesMappers = parseContext.mapperService().fullName(FieldNamesFieldMapper.NAME);
-        final FieldNamesFieldMapper fieldNamesMapper = (FieldNamesFieldMapper)fieldNamesMappers.mapper();
+        final FieldNamesFieldMapper fieldNamesMapper = fieldNamesMappers == null ? null : (FieldNamesFieldMapper)fieldNamesMappers.mapper();
         MapperService.SmartNameObjectMapper smartNameObjectMapper = parseContext.smartObjectMapper(fieldPattern);
         if (smartNameObjectMapper != null && smartNameObjectMapper.hasMapper()) {
             // automatic make the object mapper pattern

--- a/src/test/java/org/elasticsearch/search/query/ExistsMissingTests.java
+++ b/src/test/java/org/elasticsearch/search/query/ExistsMissingTests.java
@@ -39,6 +39,23 @@ import static org.elasticsearch.test.hamcrest.ElasticsearchAssertions.*;
 
 public class ExistsMissingTests extends ElasticsearchIntegrationTest {
 
+    public void testExistsFilterOnEmptyIndex() throws Exception {
+        createIndex("test");
+        ensureYellow("test");
+
+        SearchResponse resp = client().prepareSearch("test").setQuery(QueryBuilders.matchAllQuery()).setPostFilter(FilterBuilders.existsFilter("foo")).execute().actionGet();
+        assertSearchResponse(resp);
+    }
+
+    public void testMissingFilterOnEmptyIndex() throws Exception {
+        createIndex("test");
+        ensureYellow("test");
+
+        SearchResponse resp = client().prepareSearch("test").setQuery(QueryBuilders.matchAllQuery()).setPostFilter(FilterBuilders.missingFilter("foo")).execute().actionGet();
+        assertSearchResponse(resp);
+    }
+
+
     public void testExistsMissing() throws Exception {
 
         XContentBuilder mapping = XContentBuilder.builder(JsonXContent.jsonXContent)

